### DISCUSSION TITLE
534974 INFO: Appending PKG archive to custom ELF section in EXE

### DISCUSSION THREAD

**Discussion by ChewingPencilLines** [pyinstaller/pyinstaller#6536] [Help] [answered]
I can't seem to successfully create an executable on Linux.

The project directory structure is as follows, where _vtest_ is a virtual environment with only the required libraries installed:

```
test/
├──  vtest/
├──  cli.py
└──  package_a/
        ├──  c_em.py
        └──  c_po.py
```

I run pyinstaller using the command:
`pyinstaller --one-file cli.py`

where _cli.py_ is as follows:

```
from package_a.c_em import main
if __name__ == '__main__':
    main()
```

I got the following traceback:

> 534974 INFO: Appending PKG archive to custom ELF section in EXE
> Traceback (most recent call last):
>   File "/mnt/store/ttt/test/vtest/bin/pyinstaller", line 8, in <module>
>     sys.exit(run())
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/__main__.py", line 124, in run
>     run_build(pyi_config, spec_file, **vars(args))
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/__main__.py", line 58, in run_build
>     PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/building/build_main.py", line 793, in main
>     build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/building/build_main.py", line 725, in build
>     exec(code, spec_namespace)
>   File "/mnt/store/ttt/test/cli.spec", line 23, in <module>
>     exe = EXE(pyz,
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/building/api.py", line 508, in init
>     self.postinit()
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/building/datastruct.py", line 155, in postinit
>     self.assemble()
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/building/api.py", line 707, in assemble
>     retcode, stdout, stderr = exec_command_all('objcopy', '--add-section', 'pydata=%s' % append_file, self.name)
>   File "/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/compat.py", line 475, in exec_command_all
>     out, err = proc.communicate(timeout=60)
>   File "/home/ttt/anaconda3/lib/python3.8/subprocess.py", line 1028, in communicate
>     stdout, stderr = self._communicate(input, endtime, timeout)
>   File "/home/ttt/anaconda3/lib/python3.8/subprocess.py", line 1871, in _communicate
>     self._check_timeout(endtime, orig_timeout, stdout, stderr)
>   File "/home/ttt/anaconda3/lib/python3.8/subprocess.py", line 1072, in _check_timeout
>     raise TimeoutExpired(
> subprocess.TimeoutExpired: Command '('objcopy', '--add-section', 'pydata=/mnt/store/ttt/test/build/cli/cli.pkg', '/mnt/store/ttt/test/dist/cli')' timed out after 60 seconds

This is my first time using pyinstaller.
I tried looking up _"objcopy"_ but can't seem to figure out what's causing the timeout.
Can someone please help? Thank you very much!

**Comment by rokm** [comment, marked as answer]
The generated PKG is probably so large that it takes more than a minute to embed into onefile executable, resulting in the timeout error. Unfortunately the timeout seems to be hardcoded here:
https://github.com/pyinstaller/pyinstaller/blob/fcff15e6e7ab6fad135c584c2b6cc5e0f7809319/PyInstaller/compat.py#L475

So you can try increasing it in your installed PyInstaller copy (to something considerably larger, like 600), i.e.,
`/mnt/store/ttt/test/vtest/lib/python3.8/site-packages/PyInstaller/compat.py`

**Reply by rokm** [reply]
Also, can you check how large the generated PKG file (`/mnt/store/ttt/test/build/cli/cli.pkg`) actually is? 

And, what packages is your program using? (i.e., is it large because you're using something like `tensorflow`, `torch` or `scipy`, or is PyInstaller accidentally collecting your whole anaconda environment for some reason...)

**Comment by ChewingPencilLines** [comment]
Thank you very much for your answers.

I tried changing the code to increase the time out duration as you suggested and it successfully produced the cli executable.

To answer your other question, my current _cli.pkg_ is around 1.8GB atm.

EDITED: Not facing any other issues at the moment, whew!
I have some other small problems here and there I am trying to iron out before I can do a final test to ensure that everything is working as expected, but thank you so much for your help!! <3
